extensions/pkg/controller/csimigration/reconciler.go

```diff
@@ -144,7 +145,7 @@ func (r *reconciler) reconcile(ctx context.Context, cluster *extensionsv1alpha1.
 			return reconcile.Result{}, nil
 		}
 
-		_, shootClient, err := NewClientForShoot(ctx, r.client, cluster.Name, client.Options{})
+		_, shootClient, err := NewClientForShoot(ctx, r.client, cluster.Name, client.Options{}, healthcheckconfig.RESTOptions{})
```

Comment:
I agree, this looks weird when you look at csimigration.
`ClientConnectionConfiguration` contains properties for `Kubeconfig`, `AcceptContentTypes` and `ContentType` we don't want the user to change. This could be confusing.
How about moving the entire content from `extensions/pkg/controller/healthcheck/config` to `extensions/pkg/controller/apis`? This would be similar to how `pkg/...` is organised and it could be identified by its name where `HealthCheckConfig` type belongs to.

Reply:
> How about moving the entire content from `extensions/pkg/controller/healthcheck/config` to `extensions/pkg/controller/apis`? This would be similar to how `pkg/...` is organised and it could be identified by its name where `HealthCheckConfig` type belongs to.

Sounds good to me. Maybe even better to move to `extensions/pkg/apis`.

Reply:
okay, done